test/state/host.cpp

```diff
@@ -376,26 +376,28 @@ evmc::Result Host::execute_message(const evmc_message& msg) noexcept
             m_state.journal_create(msg.recipient, exists);
     }
 
-    assert(msg.kind != EVMC_CALL || evmc::address{msg.recipient} == msg.code_address);
-    auto* const dst_acc =
-        (msg.kind == EVMC_CALL) ? &m_state.touch(msg.recipient) : m_state.find(msg.code_address);
-
-    if (msg.kind == EVMC_CALL && !evmc::is_zero(msg.value))
+    if (msg.kind == EVMC_CALL)
     {
-        // Transfer value: sender → recipient.
-        // The sender's balance is already checked therefore the sender account must exist.
-        const auto value = intx::be::load<intx::uint256>(msg.value);
-        assert(m_state.get(msg.sender).balance >= value);
-        m_state.journal_balance_change(msg.sender, m_state.get(msg.sender).balance);
-        m_state.journal_balance_change(msg.recipient, dst_acc->balance);
-        m_state.get(msg.sender).balance -= value;
-        dst_acc->balance += value;
+        auto& dst_acc = m_state.touch(msg.recipient);
```

Comment:
If we send value we can actually skip touching (because the destination account will have this value or it has code to zero it later). Can you check this hypothesis (all tests passing) or at least leave a comment?